### ISSUE TITLE
Adding Plugin gear to Custom Cyberlimb Weapons

### ISSUE THREAD

**Issue by HaikenEdge** [chummer5a/chummer5a#727]
### Expected behaviour

When adding a Custom gun to a cyberlimb as a plugin, it should be possible to add an actual weapon of the type to the custom gun as a gear, to specify what the custom gun is specifically.
### Actual behaviour

When attempting to add a gear to a Custom gun in a cyberlimb as a plugin, the message "Gear cannot be added to this piece of Cyberware" appears.
### Steps to reproduce the behaviour
1. Create or open a character.
2. Under Cyberware and Bioware, add a cyberlimb.
3. Add a Custom gun to the cyberlimb as a plugin.
4. Right click on the Custom gun and select "Add Gear"

**Comment by chummer5a** [comment]
Right, let's see what can be done to get this one sorted. The easiest way for me to implement this is to add the SelectWeapon bonus to the custom weapons. That would allow you to pick from the existing weapons that the character owns, then attach the name of that weapon as an Extra value. The displayname would wind up looking like Custom Heavy Pistol (Ares Predator V).

Next simplest would be addition to the AddWeapon bonus, setting it up to purchase a weapon instead of directly adding one to the character's weapons list. Not particularly difficult, but may have some unintended effects.
